src/quo2/core.cljs

```diff
@@ -213,6 +212,12 @@
 (def markdown-list quo2.components.markdown.list.view/view)
 (def text quo2.components.markdown.text/text)
 
+;;;; MESSAGES
+(def system-message-added quo2.components.messages.system-message/system-message-added)
```

Comment:
oh please let's avoid these types of exports from quo2.core. This should be handled by the component. 
It is much easier to configure the components correctly if we just align them with Figma properties and naming conventions.
In the Figma designs there is only one component so let's only export one - this way devs do not need to figure out which one to use - it is already matching figma 

Here is it is better that we have one component and separate this with a `type` prop as it is done in Figma
<img width="571" alt="Screenshot 2023-07-27 at 11 35 26" src="https://github.com/status-im/status-mobile/assets/22799766/c1e3dad6-1982-45d4-a3af-28de8577094a">

For example we can differentiate this by `type` like this
"system-message-added quo2.components.messages.system-message/view.cljs"
```
(:require
 [quo2.components.messages.system-message :as system-message-removed]
 [quo2.components.messages.system-message-pinned :as system-message-pinned]
 [quo2.components.messages.system-message-deleted :as system-message-deleted]
 [quo2.components.messages.system-message :as system-message-added])

(defn view [{:keys [type] :as props}]
  (case type
    :added-short <type needed here>
    :added-long <type needed here>
    :contact-request <type needed here>
    :pinned-text  [system-message-pinned props]
    :deleted [system-message-deleted props]
    :deleted-and-action <type needed here>
    :pinned-gifs <type needed here>
    :pinned-photos <type needed here>))
```

Reply:
actually it was implemented with types, but its kinda weird, because they are completely different components, and just hide case inside component is kinds weird, but if we have a consensus i'm fine to change this

Reply:
yeah I agree that it is weird and sometimes leading to less "smart" code, but it is much easier to see that a component is configured correctly on the screen if we follow this approach.

For example looking at the Page Nav component being used on the communities page - 

<img src="https://github.com/status-im/status-mobile/assets/22799766/e83ba969-a00f-401c-86ee-5d8f57c1d754" width="350px" />

if the page nav component was created correctly we could then look at the code and easily see that the configuration is correct - however if “random” names for the component props were used we then would have to manually check how to get the components to match up with the designs when using them - this then leads to lots of misuses of the design system components 

There are also a tonne of our components which are not properly encapsulating styles, colors etc and have too broad an api, hopefully this approach can help to reduce that too as we align better with what is in the Design System.

Reply:
in my experience its useless, our goal as developers make code as simple and clean as possible, it should be convenient for developers, we don't care about figma, when I want to use a component I just look at the implementation in the code, not in figma, imho
regarding figma, we already tried but it didn't work, designers change a lot, and there is no consistency in naming, some designers name some do not, etc etc, so idea is good but it won't work, imho

Reply:
actually in most cases "components" is used only once, so they are not really components, they won't be used or changed in time, so we should keep ourselves calm here and don't overthink this much :)

Reply:
I would have to disagree on this. I've been refactoring a lot of components lately and the amount of misuses of components/code is mind blowing. So many components are just using overrides for color and text etc when they should be encapsulated and achieved with some configuration - imo we for the most part have not been using the design system properly as of yet.

As the project and team grows these problems will only grow further and it will become a mess to deal with.

Sure in some cases it might be a burden, however imo the likelihood of other devs using the design system correctly increases the more components aligned with this approach.

It is true that the designers can change the components but that is really a separate issue and a process that definitely needs improvement 👍

Reply:
probably for some components type can be used, but here it's useless

Reply:
@flexsurfer, there was a retrospective recently where the entire team had the chance to discuss these points and it was decided that we'll try to match much more strictly what's in the DS.

The guideline **for quo2 code** would be (is) as simple as it gets:

> Adhere to the properties and structure of the Figma component to establish the Quo2 component's folder organization, filename, and customizable properties that impact its variations.

There's no "if" or exceptions. Even for a team of our size I think it makes for an easy to follow convention. It simplifies designer's reviews and code reviews. There's no funny business, just straight out boring code.

As @J-Son89 said, it's indeed mind-blowing the amount of bugs in quo2. The hypothesis we want to verify is that this guideline will ensure we drift less from the true source of truth of the Design System, which is Figma, not the code.

If you think we should re-evaluate this decision as a team, I think it'll be more more inclusive and productive to have it outside this PR, because there are other people that could participate more easily.

-----

For an example of a component that was recently rewritten with this guideline in mind, take a look at `quo2.components.list-items.community.view` in https://github.com/status-im/status-mobile/pull/16527. The designer gave positive feedback about how easy it was to review it in the preview area when all the variations could be easily checked because all names matched. That's one of the basic points of having a DS. As a developer, I felt the same, I could more easily verify the 20+ variations in the code. And I guess reviewers could also more easily double-check things.

Reply:
thanks for the detailed explanation, somehow I didn't participate in that discussion

Reply:
Sorry @flexsurfer, I take as feedback to me as well that we should make sure decisions are not lost in the myriad of topics discussed in retros and should be communicated again in Discord, for example, so if someone was in time-off or couldn't participate, we give them a chance to also share their opinion.

As usual with new guidelines, let's give it a chance for a little while. It seems like a good direction for many of us, but we can all be wrong.